pretyping/evardefine.ml

```diff
@@ -163,7 +163,7 @@ let rec evar_absorb_arguments env evd (evk,args as ev) = function
 (* Refining an evar to a sort *)
 
 let define_evar_as_sort env evd (ev,args) =
-  let evd, u = new_univ_variable univ_rigid evd in
+  let evd, u = new_univ_variable univ_flexible_alg evd in
```

Comment:
I don't think allowing algebraic is correct here.
IIUC for instance `let s := _ in let t := _ : s in fun x : t => ...` will define_evar_as_sort on the hole for `s` so we would get an algebraic in a let body.

Reply:
Seems indeed to make sense.